source/common/memory/utils.cc

```diff
@@ -3,15 +3,26 @@
 #include "common/common/assert.h"
 #include "common/memory/stats.h"
 
-#ifdef TCMALLOC
+#if defined(TCMALLOC)
+#include "tcmalloc/malloc_extension.h"
+#elif defined(GPERFTOOLS_TCMALLOC)
 #include "gperftools/malloc_extension.h"
 #endif
 
 namespace Envoy {
 namespace Memory {
 
+namespace {
+#if defined(TCMALLOC) || defined(GPERFTOOLS_TCMALLOC)
+// TODO(zyfjeff): Make max unfreed memory byte configurable
+constexpr uint64_t MAX_UNFREED_MEMORY_BYTE = 100 * 1024 * 1024;
```

Comment:
Move this into `releaseFreeMemory`?

Reply:
It's used in two functions: `releaseFreeMemory()` and `tryShrinkHeap()`.